multichain-testing/test/kitchen-sink.test.ts

```diff
@@ -0,0 +1,169 @@
+
+    const doOffer = makeDoOffer(wdUser1);
+    t.log(`${scenario.callPipe.join('.')} offer`);
+    const offerId = `${scenario.callPipe.join('.')}-${Date.now()}`;
+
+    // FIXME we get payouts but not an offer result; it times out
+    // https://github.com/Agoric/agoric-sdk/issues/9643
+    // chain logs shows an UNPUBLISHED result
+    const _offerResult = await doOffer({
+      id: offerId,
+      invitationSpec: {
+        source: 'agoricContract',
+        instancePath: [scenario.contractName],
+        callPipe: [scenario.callPipe],
+      },
+      offerArgs: {
+        chainName: scenario.chain,
+      },
+      proposal: {},
+    });
+    t.true(_offerResult);
+    // t.is(await _offerResult, 'UNPUBLISHED', 'representation of continuing offer');
+
+    // TODO fix above so we don't have to poll for the offer result to be published
+    // https://github.com/Agoric/agoric-sdk/issues/9643
+    const currentWalletRecord = await retryUntilCondition(
+      () =>
+        vstorageClient.queryData(
+          `published.wallet.${wallets[scenario.wallet]}.current`,
+        ),
+      ({ offerToPublicSubscriberPaths }) =>
+        Object.fromEntries(offerToPublicSubscriberPaths)[offerId],
+      `${scenario.callPipe[0]} continuing invitation is in vstorage`,
+    );
+
+    const offerToPublicSubscriberMap = Object.fromEntries(
+      currentWalletRecord.offerToPublicSubscriberPaths,
+    );
+
+    const address = offerToPublicSubscriberMap[offerId]?.account
+      .split('.')
+      .pop();
+    t.log('Got address:', address);
+    t.regex(
+      address,
+      new RegExp(`^${scenario.expectedAddressPrefix}1`),
+      `address for ${scenario.chain} is valid`,
+    );
+
+    const latestWalletUpdate = await vstorageClient.queryData(
+      `published.wallet.${wallets[scenario.wallet]}`,
+    );
+    t.log('latest wallet update', latestWalletUpdate);
+    t.like(
+      latestWalletUpdate.status,
+      {
+        id: offerId,
+        numWantsSatisfied: 1,
+        result: 'UNPUBLISHED',
+        error: undefined,
+      },
+      'wallet offer satisfied without errors',
+    );
+  },
+);
+
+const chainConfigs = {
+  cosmoshub: {
+    chain: 'cosmoshub',
+    chainId: 'gaialocal',
+    denom: 'uatom',
+    expectedAddressPrefix: 'cosmos',
+    callPipe: ['makeCosmosOrchAcctInvitation'],
+    offerArgs: { chainName: 'cosmoshub' },
+  },
+  osmosis: {
+    chain: 'osmosis',
+    chainId: 'osmosislocal',
+    denom: 'uosmo',
+    expectedAddressPrefix: 'osmo',
+    callPipe: ['makeCosmosOrchAcctInvitation'],
+    offerArgs: { chainName: 'osmosis' },
+  },
+  agoric: {
+    chain: 'agoric',
+    chainId: 'agoriclocal',
+    denom: 'ubld',
+    expectedAddressPrefix: 'agoric',
+    callPipe: ['makeLocalOrchAcctInvitation'],
+  },
+};
+
+for (const [chainName, config] of Object.entries(chainConfigs)) {
+  for (const wallet of accounts) {
+    const scenario = {
+      ...config,
+      contractName: 'kitchenSink',
+      wallet,
```

Comment:
I'm curious why this is so parameterized. The macro could just take `chainName` and figure the rest of this out itself.

Reply:
Recognize some room for improvement here, maybe `chain`, `chainId`, `denom`, `expectedAddressPrefix` should come from support.ts function. But seems mostly harmless imo.

Looking forward to others contributing to this directory, and making improvements along the way.